### PR TITLE
UTOPIA-62: Setup Application Configuration via Environment Variables

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

UTOPIA-62: Setup Application Configuration via Environment Variables

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by antsand — COMMENTED** [review]
I would rename this to app. The repo already has dpia in its name.

File: src/frontend/dpia-webapp/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# dpia-webapp
+
+This template should help get you started developing with Vue 3 in Vite.
+
+## Recommended IDE Setup
```
Comment (antsand) [inline src/frontend/dpia-webapp/README.md:5]: Generally not a big fan of IDE specifics. But I also realize 99.999% use Visual studio in BC-gov so my comment is irrelevant.

File: src/frontend/dpia-webapp/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+TypeScript cannot handle type information for `.vue` imports by default, so we replace the `tsc` CLI with `vue-tsc` for type checking. In editors, we need [TypeScript Vue Plugin (Volar)](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin) to make the TypeScript language service aware of `.vue` types.
+
+If the standalone TypeScript plugin doesn't feel fast enough to you, Volar has also implemented a [Take Over Mode](https://github.com/johnsoncodehk/volar/discussions/471#discussioncomment-1361669) that is more performant. You can enable it by the following steps:
+
+1. Disable the built-in TypeScript Extension
+    1) Run `Extensions: Show Built-in Extensions` from VSCode's command palette
+    2) Find `TypeScript and JavaScript Language Features`, right click and select `Disable (Workspace)`
+2. Reload the VSCode window by running `Developer: Reload Window` from the command palette.
+
+## Customize configuration
```
Comment (antsand) [inline src/frontend/dpia-webapp/README.md:20]: For now this is fine. I want to have a conversation with our directors. The frontend needs to be more modular. We need to define an architecture. 

To give context, The whole application is modular with micro-services. But the front-end has 1 point of failure. The frontend is a SPA bundled into one javascript file. If a dependencies or version upgrade is necessary and it breaks the app, the whole frontend app will stop working.  

The majority of the code will be front-end so this needs to be thought out properly.

File: src/frontend/dpia-webapp/index.html [hunk 3]
```diff
@@ -8,6 +8,6 @@
   </head>
   <body>
     <div id="app"></div>
-    <script type="module" src="/src/main.js"></script>
+    <script type="module" src="/src/main.ts"></script>
```
Comment (antsand) [inline src/frontend/dpia-webapp/index.html:11]: Why is this main.ts? Is not typescript compiled to .js? Has this been tested?

File: src/frontend/dpia-webapp/src/assets/main.css [hunk 4]
```diff
@@ -0,0 +1,35 @@
+@import "./base.css";
```
Comment (antsand) [inline src/frontend/dpia-webapp/src/assets/main.css:1]: Should we use Sass for this ?

File: src/frontend/dpia-webapp/src/components/TheWelcome.vue [hunk 5]
```diff
@@ -1,10 +1,10 @@
-<script setup>
-import WelcomeItem from './WelcomeItem.vue'
```
Comment (antsand) [inline src/frontend/dpia-webapp/src/components/TheWelcome.vue:2]: These files are default files when we create starter vue project. Eventually we would need to delete them. Such an architecture by default calls for single point of failure.

**Review by antsand — COMMENTED** [review]
Few comments. You can commit the code. We can address each concern over time.

File: src/frontend/dpia-webapp/index.html (hunk 3 above)
Comment (antsand) [reply src/frontend/dpia-webapp/index.html:11]: test frontend before committing. the main file index.html has a main.ts (typescript). I am not sure if the vue compiler will reformat .ts to .js on index.html. Browsers do not understand .ts files.

Comment (akroon3r) [reply src/frontend/dpia-webapp/index.html:11]: This has to do with project init and me choosing whether or not vue has typescript support. Try doing a project init on your local outside of the repo and you can see. This diff is generated by me originally choosing no typescript support over typescript support.

File: src/frontend/dpia-webapp/src/assets/main.css (hunk 4 above)
Comment (akroon3r) [reply src/frontend/dpia-webapp/src/assets/main.css:1]: this is how vue init's a project, this import wasn't generated by me. we can look into changing this in the future if we require.

File: src/frontend/dpia-webapp/src/components/TheWelcome.vue (hunk 5 above)
Comment (akroon3r) [reply src/frontend/dpia-webapp/src/components/TheWelcome.vue:2]: It's part of the hello world you can see served in openshift. We can delete the components when we actually start development.

**Review by antsand — APPROVED** [review]
Works for now.

**Pull request merged**